lib/utils/domFns.es6

```diff
@@ -127,21 +127,24 @@ export function getTouchIdentifier(e: MouseTouchEvent): ?number {
 // User-select Hacks:
 //
 // Useful for preventing blue highlights all over everything when dragging.
-const userSelectPrefix = getPrefix('user-select');
-const userSelect = browserPrefixToStyle('user-select', userSelectPrefix);
-const userSelectStyle = `;${userSelect}: none;`;
-const userSelectReplaceRegExp = new RegExp(`;?${userSelect}: none;`); // leading ; not present on IE
 
 // Note we're passing `document` b/c we could be iframed
-export function addUserSelectStyles(body: HTMLElement) {
-  const style = body.getAttribute('style') || '';
-  if (userSelectReplaceRegExp.test(style)) return; // don't add twice
-  body.setAttribute('style', style + userSelectStyle);
+export function addUserSelectStyles(doc: Document) {
+  let styleEl = doc.getElementById('react-draggable-style-el');
+  if (!styleEl) {
+    styleEl = doc.createElement('style');
+    styleEl.type = 'text/css';
+    styleEl.id = 'react-draggable-style-el';
+    styleEl.innerHTML = '.react-draggable-transparent-selection *::-moz-selection {background: transparent;}\n';
+    styleEl.innerHTML += '.react-draggable-transparent-selection *::selection {background: transparent;}\n';
+    doc.getElementsByTagName('head')[0].appendChild(styleEl);
+  }
+  addClassName(doc.body, 'react-draggable-transparent-selection'); 
 }
 
-export function removeUserSelectStyles(body: HTMLElement) {
-  const style = body.getAttribute('style') || '';
-  body.setAttribute('style', style.replace(userSelectReplaceRegExp, ''));
+export function removeUserSelectStyles(doc: Document) {
+  removeClassName(doc.body, 'react-draggable-transparent-selection');
+  window.getSelection().removeAllRanges();  // remove selection caused by scroll    
```

Comment:
@STRML @dherbolt Can you please tell me how can scrolling cause selection? And why this removal is important?

Reply:
When you hold mouse button down to drag some element it cases also invisible (transparent) selection of elements on the page. When you reach bottom/top border of the browser window, this selection starts scrolling with the content in the window (as when you are selecting text).
When you leave mouse button, this selection used to scroll with the page should be removed otherwise it will be probably visible in some cases or can cause somethig else. It isn't nothing else then final cleanup.

Reply:
Hey @dherbolt,
Sorry to ping on something old but, does that mean the auto-scroll feature relies on the hidden selection ?

Reply:
Yes, exactly.